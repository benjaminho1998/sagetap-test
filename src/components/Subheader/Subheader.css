.subheader-container {
    height: 60px;
    background-color: teal;
    color: white;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 20px;
}