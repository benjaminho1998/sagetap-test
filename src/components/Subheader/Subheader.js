import React from 'react';
import './Subheader.css';
import CommentIcon from '@mui/icons-material/Comment';
import Badge from '@mui/material/Badge';
import IconButton from '@mui/material/IconButton';
import Tooltip from '@mui/material/Tooltip';

const Subheader = (props) => {

    const handleShow = () => {
        props.handleShow();
    }

    return (
        <div className='subheader-container'>
            <span>Demo Project</span>
            <Tooltip title='Open Comments'>
                <IconButton onClick={handleShow} aria-label='comments'>
                    <Badge badgeContent={props.commentsLength} color='error'>
                        <CommentIcon style={{ color: "white" }} color='action' />
                    </Badge>
                </IconButton>
            </Tooltip>
        </div>
    );
}

export default Subheader;