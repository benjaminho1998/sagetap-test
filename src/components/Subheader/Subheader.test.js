import React from 'react';
import IconButton from '@mui/material/IconButton';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Subheader from './Subheader';

describe('Subheader Component', () => {
  it('should call handleShow', () => {
    const handleShowSpy = jest.fn();
    render(
        <IconButton onClick={handleShowSpy} />
    );
    fireEvent.click(screen.queryByRole('button'));
    expect(handleShowSpy).toHaveBeenCalled();
  });

  it('renders snapshot correctly with props', () => {
    const handleShow = jest.fn();
    const tree = renderer.create(<Subheader handleShow={handleShow} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
