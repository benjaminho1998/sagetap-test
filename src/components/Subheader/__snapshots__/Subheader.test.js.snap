// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subheader Component renders snapshot correctly with props 1`] = `
<div
  className="subheader-container"
>
  <span>
    Demo Project
  </span>
  <button
    aria-label="comments"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiBadge-root css-1c32n2y-MuiBadge-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-11pbyhm-MuiSvgIcon-root"
        data-testid="CommentIcon"
        focusable="false"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
        />
      </svg>
      <span
        className="MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-invisible css-zkqb44-MuiBadge-badge"
      />
    </span>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
